src/TestRunnerOrchestrator.ts

```diff
@@ -43,16 +43,24 @@ export default class TestRunnerOrchestrator {
           return null; // we're done
         } else {
           var mutant = mutants.pop();
-          let nextRunner = testRunners.pop();
-          let sourceFileCopy = nextRunner.sourceFileMap[mutant.filename];
-          return Promise.all([mutant.save(sourceFileCopy), nextRunner.selector.select(mutant.scopedTestIds)])
-            .then(() => nextRunner.runnerAdapter.run({ timeout: this.calculateTimeout(mutant.timeSpentScopedTests) }))
-            .then((runResult) => {
-              this.updateMutantStatus(mutant, runResult);
+
+          if (mutant.scopedTestIds.length > 0) {
+            let nextRunner = testRunners.pop();
+            let sourceFileCopy = nextRunner.sourceFileMap[mutant.filename];
+            return Promise.all([mutant.save(sourceFileCopy), nextRunner.selector.select(mutant.scopedTestIds)])
+              .then(() => nextRunner.runnerAdapter.run({ timeout: this.calculateTimeout(mutant.timeSpentScopedTests) }))
+              .then((runResult) => {
+                this.updateMutantStatus(mutant, runResult);
+                reporter.mutantTested(mutant);
+                return mutant.reset(sourceFileCopy);
+              })
+              .then(() => testRunners.push(nextRunner)); // mark the runner as available again
+          } else {
+            return new Promise<number>(resolve => {
               reporter.mutantTested(mutant);
-              return mutant.reset(sourceFileCopy);
```

Comment:
This can be easier:
`return Promise.resolve(() => reporter.mutantTested(mutant));`